src/components/CRUD.tsx

```diff
@@ -0,0 +1,542 @@
+    if (!connection(true)) {
+      return;
+    }
+    try {
+      Keyboard.dismiss();
+      richTextRef.current?.blurContentEditor();
+      setLoading(true);
+      let response;
+
+      const htmlValue = richHTML?.replace(/<div><br><\/div>/g, '') === '' ? '' : richHTML;
+      if (type === 'thread') {
+        if (title) {
+          if (action === 'create' && forumId) {
+            if (!htmlValue) {
+              throw Error('First post cannot be empty.');
+            }
+            response = await createThread(title, htmlValue, forumId).request;
+          } else if (threadId) {
+            if (thread) {
+              dispatch(updateThreads({ ...thread, title: title }));
+            }
+            response = await updateThread(threadId, { title: title }).request;
+          }
+        } else {
+          throw Error('Title cannot be empty.');
+        }
+      } else {
+        if (htmlValue) {
+          const content = `${
+            quotes && quotes?.length > 0
+              ? quotes
+                  ?.map(({ content: mapContent }) => mapContent)
+                  .join('<br>')
+                  .concat('<br>')
+              : ''
+          }${htmlValue}`;
+
+          if (action === 'create' && threadId) {
+            response = await createPost({
+              content,
+              thread_id: threadId,
+              parent_ids: (quotes as IPost[])?.map(q => q?.id),
+            }).request;
+          } else if (postId) {
+            if (post) {
+              dispatch(updatePosts({ ...post, content }));
+            }
+            response = await editPost(postId, content).request;
+          }
+        } else {
+          throw Error('Post cannot be empty.');
+        }
+      }
+
+      setLoading(false);
+      onPostCreated?.(response?.data?.id);
+      if (type === 'thread' && action === 'create') {
+        pop();
+        navigate('Thread', { threadId: response?.data?.id });
+      } else {
+        goBack();
+      }
+    } catch (e: any) {
+      customModalRef.current?.toggle(
+        'Something went wrong',
+        e?.errors?.map((m: { detail: any }) => m?.detail).join(' ') ||
+          e?.message ||
+          'Please try again later.'
+      );
+      setLoading(false);
+    }
+  }, [
+    richHTML,
+    title,
+    action,
+    dispatch,
+    forumId,
+    goBack,
+    navigate,
+    onPostCreated,
+    pop,
+    post,
+    postId,
+    quotes,
+    thread,
+    threadId,
+    type,
+  ]);
+
+  const onDelete = useCallback(async () => {
+    if (!connection(true)) {
+      return;
+    }
+    Keyboard.dismiss();
+    setLoading(true);
+    if (type === 'thread' && threadId) {
+      await deleteThread(threadId).request;
+      pop();
+    } else if (postId) {
+      await deletePost(postId).request;
+      onDeleteProp?.(postId);
+    }
+    goBack();
+  }, [goBack, pop, postId, threadId, type, onDeleteProp]);
+
+  return (
+    <SafeAreaView
```

Comment:
Something is off here too, take a look how this page should look when replying to a post
![image](https://github.com/railroadmedia/react-native-musora-forum/assets/32734435/bc6a4584-c126-412a-9cfd-9b3c531d9c5b)
When I test this with your changes  the original post to which I'm replying is not showing

Reply:
Fixed it